lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1069,13 +1078,74 @@ SPIRVValue *SPIRVModuleImpl::addNullConstant(SPIRVType *Ty) {
 
 SPIRVValue *SPIRVModuleImpl::addCompositeConstant(
     SPIRVType *Ty, const std::vector<SPIRVValue *> &Elements) {
-  return addConstant(new SPIRVConstantComposite(this, Ty, getId(), Elements));
+  const int MaxNumElements = MaxWordCount - SPIRVConstantComposite::FixedWC;
+  const int NumElements = Elements.size();
+
+  // In case number of elements is greater than maximum WordCount and
+  // SPV_INTEL_long_constant_composite is not enabled, the error will be emitted
+  // by validate functionality of SPIRVCompositeConstant class.
+  if (NumElements <= MaxNumElements ||
+      !isAllowedToUseExtension(ExtensionID::SPV_INTEL_long_constant_composite))
+    return addConstant(new SPIRVConstantComposite(this, Ty, getId(), Elements));
```

Comment:
It seems strange to me that here we might intentionally create "invalid" SPIR-V instruction and it will be checked second time later to emit an error. I would prefer not to create a `SPIRVConstantComposite` with huge amount of continuents if the extension is disabled and just emit an error right away.

However, this seems more like a refactoring, so not asking to do this right now

Reply:
Add error checking to tests.